picasso/src/main/java/com/squareup/picasso3/NetworkRequestHandler.kt

```diff
@@ -67,8 +68,14 @@ internal class NetworkRequestHandler(
             picasso.downloadFinished(body.contentLength())
           }
           try {
-            val bitmap = decodeStream(body!!.source(), request)
-            callback.onSuccess(Result.Bitmap(bitmap, loadedFrom))
+            val bufferedSource = body!!.source()
+            if (Utils.isGifFile(bufferedSource)) {
+              val drawable = decodeDrawableStream(bufferedSource, request)
+              callback.onSuccess(Result.Drawable(drawable, loadedFrom))
+            } else {
+              val bitmap = decodeStream(bufferedSource, request)
+              callback.onSuccess(Result.Bitmap(bitmap, loadedFrom))
+            }
```

Comment:
Do you think it would be more readable like this?

```suggestion
            val result = if (Utils.isGifFile(bufferedSource)) {
              val drawable = decodeDrawableStream(bufferedSource, request)
              Result.Drawable(drawable, loadedFrom)
            } else {
              val bitmap = decodeStream(bufferedSource, request)
              Result.Bitmap(bitmap, loadedFrom)
            }
            callback.onSuccess(result)
```

Also, can `isGifFile` be static imported here?

Reply:
👍 I like extracting the result and don't have a strong preference with the import. Looking around the code base it seems mixed with doing the full static import